Implement an active-only-sessions view, per request on Github 3831 (RM 5844)

cdwiegand opened the pull request:
This has been bugging me for years, I just wanted to only show active sessions sometimes to then filter/sort by database, and identify my problem queries quickly.

I took feedback from prior PR, changed from a new tab to a checkbox within the CustomHeader of PgTable. Idea is to allow users to only show non-Idle connections, so we can then sort by other columns (e.g. database, connection time, etc.) without cluttering our display with Idle connections resulting from connection poolers or some ORMs/libaries.

adityatoshniwal commented:
@cdwiegand Excellent work.
A question, shouldn't we simply filter active sessions in the front end data itself? That way we'll not need an extra API to manage and we can get switch between active or all session instantaneously by just using the checkbox.

cdwiegand commented:
I don't see a way for me to filter that in the <PgTable> using a JavaScript function - is there an example of that in pgadmin? That's why I used an API endpoint to do it. I tried using query strings but flask wouldn't pass it.

adityatoshniwal commented on web/pgadmin/dashboard/static/js/Dashboard.jsx:
```diff
@@ -833,6 +839,26 @@ export default function Dashboard({
+      ></InputCheckbox>);
+  };
+    
   return (
```
To filter on front end:
```
const finalDashData = useMemo(()=>{
  if(tabVal == 0 && activeOnly) {
    return dashData.filter((r)=>(r.state=='active'));
  }
  return dashData;
}, [activeOnly, tabVal])
```
and pass finalDashData to all the tables.

cdwiegand replied:
this helped immensely, a little more digging and I had it working well. I ended up adding dashData to the array of monitored objects as I couldn't tell if that would inherently trigger (and my browser seems to not clear cache reliably, so it was going to take a long time to determine it manually).

adityatoshniwal commented on web/pgadmin/dashboard/static/js/Dashboard.jsx:
```diff
@@ -833,6 +839,26 @@ export default function Dashboard({
+        onChange={(e) => {
+          e.preventDefault();
+          setActiveOnly(e.target.checked);
+          setRefresh(!refresh);
```
You'll not need setRefresh once the active only is filtered at frontEnd;

adityatoshniwal commented on web/pgadmin/dashboard/static/js/Dashboard.jsx:
```diff
@@ -833,6 +839,26 @@ export default function Dashboard({
     );
   };
 
+  const CustomActiveOnlyHeaderLabel =
```
No need to create variable. Pass the object directly

cdwiegand replied:
Yeah I tried that on my side, and I could not get that to work reliably for me, which is why I ended up creating it as a variable so that I could get getText() to work correctly. I will review these changes, incorporate them into my branch, and then update my pull request. Thank you for your help with this.

adityatoshniwal commented on web/pgadmin/dashboard/__init__.py:
```diff
@@ -353,12 +357,26 @@ def get_data(sid, did, template, check_long_running_query=False):
     )
 
 
+def filter_active_only(rows):
```
As suggested, lets filter active only at front end;

cdwiegand commented:
I've updated my branch and squashed the commit/force pushed to update to only one commit, it works for me from a clean browser to check or uncheck and handle it entirely client-side.

adityatoshniwal commented:
@cdwiegand Very nice! The changes looks good to me. Will run the git workflows and merge if all good.
